src/Fixer/Operator/OperatorLinebreakFixer.php

```diff
@@ -0,0 +1,229 @@
+use PhpCsFixer\FixerConfiguration\FixerConfigurationResolver;
+use PhpCsFixer\FixerConfiguration\FixerOptionBuilder;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Preg;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Kuba Werłos <werlos@gmail.com>
+ */
+final class OperatorLinebreakFixer extends AbstractFixer implements ConfigurationDefinitionFixerInterface
+{
+    /**
+     * @internal
+     */
+    const BOOLEAN_OPERATORS = [
+        '&&',
+        '||',
+        'and',
+        'or',
+        'xor',
+    ];
+
+    /**
+     * @internal
+     */
+    const NON_BOOLEAN_OPERATORS = [
+        '=',
+        '*',
+        '/',
+        '%',
+        '<',
+        '>',
+        '|',
+        '^',
+        '+',
+        '-',
+        '&',
+        '&=',
+        '.=',
+        '/=',
+        '=>',
+        '==',
+        '>=',
+        '===',
+        '!=',
+        '<>',
+        '!==',
+        '<=',
+        '-=',
+        '%=',
+        '*=',
+        '|=',
+        '+=',
+        '<<',
+        '<<=',
+        '>>',
+        '>>=',
+        '^=',
+        '**',
+        '**=',
+        '<=>',
+        '??',
+    ];
+
+    /** @var string[] */
+    private $operators;
+
+    /** @var string */
+    private $position = 'beginning';
+
+    public function __construct()
+    {
+        $this->operators = \array_merge(self::BOOLEAN_OPERATORS, self::NON_BOOLEAN_OPERATORS);
+
+        parent::__construct();
+    }
+
+    /**
+     * @return FixerDefinition
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Binary operators must always be at the beginning or at the end of the line.',
+            [new CodeSample('<?php
+function foo() {
+    return $bar ||
+        $baz;
+}
+')]
+        );
+    }
+
+    /**
+     * @return FixerConfigurationResolver
+     */
+    public function getConfigurationDefinition()
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('only_booleans', 'whether to limit operators to only boolean ones'))
+                ->setDefault(false)
+                ->setAllowedTypes(['bool'])
+                ->getOption(),
+            (new FixerOptionBuilder('position', 'whether to place operators at the beginning or at the end of the line'))
+                ->setDefault($this->position)
+                ->setAllowedValues(['beginning', 'end'])
+                ->getOption(),
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configure(array $configuration = null)
+    {
+        parent::configure($configuration);
+
+        if (isset($configuration['only_booleans']) && $configuration['only_booleans']) {
+            $this->operators = self::BOOLEAN_OPERATORS;
+        }
+        $this->position = isset($configuration['position']) ? $configuration['position'] : $this->position;
```

Comment:
What's the use of setting the property's default if you're overriding it here anyway, optionally with its own value?

Reply:
Also, it has a default value, doesn't that mean that it's always set anyway?